projects/nexton/index.js

```diff
@@ -0,0 +1,35 @@
+
+const { call, getTokenRates } = require("../helper/chain/ton");
+
+const contractAddress = "EQBED_4VLQC-SRdrVCtmcmwO7PtnvdLSRFRYNaGIZBqpaQHQ";
+const nxton = {
+    token: "0:9d123d5d121efaf9069c737f19447d9e1adab2a7ee747cedffd3004a17c1ca4f",
+    name: "NxTON",
+    decimals: 9,
+};
+
+const getTVL = async (api)=> {
+    const tonStaked = (await call({
+        target: contractAddress,
+        abi: "tonStaked"
+    }))[0];
+    const nxtonStaked = (await call({
```

Comment:
how are nxton minted? it is not very clear to me

Reply:
Nxton is the liquid staking token of our service. It can also be used for restaking.

Reply:
ah, then we should not count it as we are already counting the TON backing it as tvl

Reply:
nxTON will be listed on Ston.fi DEX this week.
There will be multiple ways to acquire nxTON besides lending, 
 - ston.fi swap, lp. 
 - earning it as rewards from partner DeFi projects, 
so it’s not just a simple converted TON. 
nxTON has its own value and more use cases coming.

Reply:
yes but are nxTON tokens minted? if we are already counting the tokens backing nxTON as tvl, we cant count nxTON as tvl

Reply:
**Here's comment from our CEO:** 

nxTON is based on TON but is not merely a representation of its underlying asset. It is issued through a lending process where NFTs are collateralized and holds independent liquidity and utility within the ecosystem. As a distinct asset from TON, it must be calculated separately in TVL. nxTON generates new economic value and provides clear, non-overlapping contributions within the protocol's structure.

Nexton’s Structural Overview
Nexton is a protocol combining TON deposits, NFT minting, and nxTON issuance with liquid staking and arbitrage operations. This structure allows for efficient asset utilization, enabling TON and nxTON to contribute separately to financial activities and ecosystem growth.

1. Key Steps in the Protocol

(1) TON Deposit and External Liquid Staking

When users deposit TON into Nexton, the TON is staked in an external liquid staking protocol (e.g., stTON).
The Liquid Staking Token (LST) generated by this process is then utilized by arbitrage bots.
Arbitrage bots create profits by leveraging LST or swapped TON, with part of the profits reinvested into the protocol.

(2) NFT Minting and nxTON Issuance

Users mint NFTs that represent their TON staking positions.
These minted NFTs act as unique assets that can be collateralized to lend nxTON.
Since nxTON is issued through a lending process, it is not directly tied to TON but instead arises from a distinct financial activity backed by collateralized NFTs.

(3) nxTON Utility

nxTON is used in various ways within the ecosystem:
TON-nxTON Liquidity Pool: Providing liquidity on DEXs like [ston.fi](http://ston.fi/).
nxTON Staking Pool: Depositing nxTON into staking pools to receive arbitrage profits and airdrop points.
Transactional and Financial Activities: nxTON serves as collateral or a trading asset in both internal and external protocols.

(4) Profit Distribution and Airdrops

nxTON holders benefit from additional profit distributions and airdrop points beyond those available from TON staking or liquid staking LSTs.

Justification for Including nxTON in TVL

1. nxTON Is Not a Simple Derivative of TON
nxTON is not a simple derivative token of TON's underlying value but instead generates independent financial utility and economic value through its unique position in the protocol.

(1) NFT Collateralization and Lending Process

nxTON is not issued directly from TON but through the collateralization of NFTs minted from deposited TON.
This process makes nxTON a new financial asset, independent of TON’s intrinsic value.

Example Workflow:

A user deposits TON and mints an NFT representing their staking position.
The NFT is used as collateral to lend nxTON.
nxTON then acts as an independent, collateral-backed asset.

(2) Independent Utility in the Ecosystem

nxTON is deployed in ways TON alone cannot be:
Liquidity Pools: nxTON pairs with TON to provide liquidity, contributing additional economic activity.
Staking: nxTON can be staked for arbitrage rewards and other benefits separate from TON's staking activities.

2. DeFi TVL Definition: Utility-Based Evaluation
In DeFi, TVL measures not only the value of locked assets but also how those assets are utilized and contribute to the ecosystem.

(1) nxTON’s Independent Contribution

nxTON is minted through collateralization and lending, making it distinct from TON.
It enables unique activities like liquidity provision, staking, and arbitrage participation that create new economic value beyond TON’s usage.

(2) Prevention of Double-Counting in TVL

nxTON is not included in TVL as a representation of TON’s value. Instead, it reflects the collateralization of NFTs and the protocol’s financial operations.
This ensures that nxTON’s TVL contribution is separate and non-overlapping with TON.

3. nxTON’s Independence in the Protocol Structure

(1) Separation of TON and nxTON

TON contributes to TVL through liquid staking and arbitrage operations.
nxTON, however, is issued through a separate process of NFT collateralization and lending, making it an independent contributor to the ecosystem.

(2) nxTON’s Role in Ecosystem Circulation

nxTON circulates independently of TON, supporting activities like:
Liquidity Supply: nxTON pairs with TON in DEX liquidity pools, providing additional TVL contributions.
Staking: nxTON holders can stake their assets to receive rewards unavailable through direct TON staking.
Collateral and Trading: nxTON is used as collateral or in transactions, enabling financial operations distinct from TON.

Counterpoint: Why nxTON Must Be Counted Separately in TVL
The claim that “if we are already counting the tokens backing nxTON as TVL, we can't count nxTON as TVL” assumes that nxTON is a direct derivative of TON. However, the structure of Nexton demonstrates that nxTON is a distinct asset with independent contributions, justifying its separate inclusion in TVL.

1. nxTON Is Not Issued from TON’s Underlying Value

nxTON is created through NFT collateralization and lending, not directly from TON deposits.
TON contributes to TVL via liquid staking and arbitrage, while nxTON arises from the protocol’s financial activities.

2. TVL Inclusion Based on Independent Utility

nxTON enables economic contributions that TON alone cannot, such as unique liquidity provision and staking opportunities.
These contributions make nxTON a separate and independent asset in the ecosystem’s TVL calculations.

Appendix: Nexton’s Expansion Potential and nxTON’s Role
Nexton’s current design focuses on TON deposits and NFT-based nxTON issuance, but it is built with scalability in mind. This further reinforces nxTON’s independence from TON.

1. Expanding Support for Additional TON LSTs and Jettons
Nexton plans to support deposits of various TON Liquid Staking Tokens (LSTs) and Jetton-based assets in the future.

(1) Deposits Beyond TON

Users will be able to deposit stTON (from liquid staking), Jetton-based stablecoins, or other TON ecosystem assets to mint NFTs.
These NFTs, when collateralized, allow for nxTON issuance, proving nxTON is not bound solely to TON as an underlying asset.

(2) Cross-Asset Interactions

Assets like stTON or Jettons can be minted into NFTs and then converted into nxTON for ecosystem use.
This allows nxTON to facilitate activities beyond what TON can achieve, such as diversified liquidity and staking.

2. Differentiation in TON Deposit and Staking Structure
While TON deposits currently operate directly within the protocol, future integration with liquid staking protocols will enhance its utility and independence.

(1) Tokenization of Staking Positions

Deposited TON will not be immediately staked into liquid staking protocols to issue nxTON. Instead, NFTs will represent staking positions for greater flexibility and liquidity.
nxTON, therefore, is not a derivative of TON but a financial product backed by collateralized positions.

(2) Revenue-Backed Value for nxTON

nxTON is not backed by TON’s value but by the protocol’s arbitrage and staking revenue.
This ensures nxTON’s value stems from economic activity within the protocol, reinforcing its independence.

Conclusion
nxTON will maintain its current structure of being issued via TON deposits and NFT collateralization while expanding to support various TON LSTs and Jetton-based assets. It is not directly backed by TON’s underlying value but by the protocol’s revenue generation. This makes nxTON an asset distinct from TON, providing unique contributions to the ecosystem and justifying its separate inclusion in TVL under DeFi standards.

Reply:
sorry, we cant count nxTON as tvl, it is textbook definition of doublecounting.